src/api.ts

```diff
@@ -5,11 +5,74 @@ import { ChildProcessWithoutNullStreams, spawn } from 'child_process';
 import { globalContext } from './extension';
 import { tickWaitlistCounter } from './waitlist';
 
-type Targets = Record<'pod' | 'deployment' | string, string[]> & { 
-    lastTarget: string | undefined;
-    length: number;
+const TARGET_DISPLAY: Record<string, string> = {
+    pod: 'Pod',
+    deployment: 'Deployment',
+    rollout: 'Rollout',
 };
 
+type TargetsInner = Record<string, string[] | undefined>;
+
+export class Targets {
+    private _activeIndex: number;
+    private _pageSwitchOptions: [string, number][] = [];
+
+    readonly keys: string[];
+    readonly inner: TargetsInner;
+    readonly length: number;
+
+    constructor(targets: string[], lastTarget?: string) {
+        this.length = targets.length;
+
+        this.inner = targets.reduce((acc, target) => {
+            const targetKey = target.split('/')[0];
+
+            if (Array.isArray(acc[targetKey])) {
+                acc[targetKey]!.push(target);
+            } else {
+                acc[targetKey] = [target];
+            }
+
+            return acc;
+        }, {} as TargetsInner);
+
+        this.keys = Object.keys(this.inner);
+
+        if (!!lastTarget) {
+            this._activeIndex = Math.max(0, this.keys.findIndex((key) => lastTarget.startsWith(key)));
+        } else {
+            this._activeIndex = Math.max(0, this.keys.indexOf('pod'));
```

Comment:
I think activeIndex can just be 0 always

Reply:
There is a test that looks for pods should I change the test?

Reply:
I think so